hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateTopLevelDagVisitor.java

```diff
@@ -236,14 +240,21 @@ public Vertex onFullScan(FullScanPhysicalRel rel) {
 
         dagBuildContext.setTable(table);
         dagBuildContext.setRel(rel);
-        return getJetSqlConnector(table).fullScanReader(
+
+        List<Map<String, Expression<?>>> partitionStrategyCandidate = null;
+        if (partitionStrategyCandidates != null) {
```

Comment:
scan pruning will not be used when member pruning is not possible. How you plan to fix that?

Reply:
Not sure if it is possible at all...

Reply:
this was a requirement. 

>Not sure if it is possible at all

It is: 6f4aa7cb464fc2b697180e9df2828b6f58c3eab2. This also demonstrates one of the limitations of current design, so it is not pretty, in wrong place and relies on some internals.

Reply:
Ok, I got it : if you have only one prunable input for, let's say, UNION, and the second is not prunable -- member pruning is not applicable, but partition pruning is. Sorry, my bad.

Reply:
this is one example, also interesting are joins: 8519c1c279df9f22c14868b4b497e35df5eabfc9